src/contracts/factory/FundFactory.ts

```diff
@@ -31,6 +80,167 @@ export class FundFactory extends Contract {
       args.version,
     ]);
   }
+
+  private componentExists(component: Address) {
+    return !isZeroAddress(component);
+  }
+
+  private validateComponentNotSet(component: Address) {
```

Comment:
I would make this helper function accept the return object of `getManagersToRoutes` and the name of the key to check. That way, you can also include the key as a value in the error object. Also call it `validateRouteNotSet` Example:

```typescript
private validateRouteNotSet(routes: FundFactoryRoutes, route: keyof FundFactoryRoutes) {
  if (!isZeroAddress(routes[route])) {
    throw new RouteNotSetError(route);
  }
}
```

Reply:
Basically, I find this too generic currently. It should be specifically about *routes* and specifically WHICH route. You should not use this method to also validate other addresses and if they are zero. E.g. for checking if the fund setup is completed or not, use a different, more specific one.